opl/opl_sdl.c

```diff
@@ -349,10 +349,7 @@ static int OPL_SDL_Init(unsigned int port_base)
     callback_mutex = SDL_CreateMutex();
     callback_queue_mutex = SDL_CreateMutex();
 
-    // Set postmix that adds the OPL music. This is deliberately done
-    // as a postmix and not using Mix_HookMusic() as the latter disables
-    // normal SDL_mixer music mixing.
-    Mix_RegisterEffect(MIX_CHANNEL_POST, OPL_Mix_Callback, NULL, NULL);
+    Mix_HookMusic(OPL_Mix_Callback, NULL);
```

Comment:
Why can't e keep this?

Reply:
This bug is fixed, mixing works (also see `i_flmusic.c`). `Mix_RegisterEffect` is buggy with a hot swapping for some reason.

Reply:
Wait, just tested it with SIGIL_SHREDS.wad, `Mix_HookMusic` actually disabled ogg/mp3 music mixing. This requires more work.

Reply:
> To disable this callback (and thus reenable built-in music playback) call this function with a NULL callback.

https://wiki.libsdl.org/SDL_mixer/Mix_HookMusic

Reply:
We already do that here: https://github.com/fabiangreffrath/woof/blob/84aa5955d0b5c0233fa71e5ff90e7fb6556b76f9/opl/opl_sdl.c#L228

I sort of fixed this issue here: https://github.com/fabiangreffrath/woof/pull/824/commits/480e9acf7d22a731fb56187d3e08fa50184d3046

Reply:
> I sort of fixed this issue here: 

Alright, but what do you mean with "sort of"?

Reply:
> Alright, but what do you mean with "sort of"?

It's not pretty, but it works well. 

I believe that this PR is ready.